vim25/debug/debug.go

```diff
@@ -54,28 +54,12 @@ func Flush() {
 	currentProvider.Flush()
 }
 
-// FileProvider implements a debugging provider that creates a real file for
-// every call to NewFile. It maintains a list of all files that it creates,
-// such that it can close them when its Flush function is called.
-type FileProvider struct {
-	Path string
-
-	files []*os.File
-}
-
-func (fp *FileProvider) NewFile(p string) io.WriteCloser {
-	f, err := os.Create(path.Join(fp.Path, p))
-	if err != nil {
-		panic(err)
+func Scrub(in []byte) []byte {
+	out := string(in)
+	if strings.Contains(out, "<password>") {
+		var re = regexp.MustCompile(`<password>(.*)</password>`)
```

Comment:
This can be moved to a package-level var so its only compiled once.

Reply:
yup great point